File: leetcode/algorithms/tree/binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def getHeight(self,root):
        if root==None:
            return 0;
        return 1 + max(self.getHeight(root.left),self.getHeight(root.right))
    
    def populateLevel(self, root, levelIndex, level):
        if root==None:
            return;
        level[levelIndex].append(root.val);
        self.populateLevel(root.left, levelIndex + 1,level);
        self.populateLevel(root.right,levelIndex + 1,level);
    
    def levelOrder(self, root):
        """
        :type root: TreeNode
        :rtype: List[List[int]]
        """
        
        treeHeight = self.getHeight(root);
        
        level = [[] for i in range(treeHeight)];
        
        self.populateLevel(root,0,level)
        
        return level;

File: leetcode/algorithms/tree/binary-tree-level-order-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    
    private void populateLevelOrderTrav(TreeNode root, int level, List<List<Integer>> levelOrderTravList) {
         if(root==null){
             return;
         }
        
        levelOrderTravList.get(level).add(root.val);
        populateLevelOrderTrav(root.left,level+1,levelOrderTravList);
        populateLevelOrderTrav(root.right,level+1,levelOrderTravList);
        
    }
    
    private int getHeight(TreeNode root) {
        if(root==null){
            return 0;
        }
        
        return 1 + Math.max(getHeight(root.left),getHeight(root.right));
        
    }
    
    public List<List<Integer>> levelOrder(TreeNode root) {
        
        int height = getHeight(root);
        
        List<List<Integer>> levelOrderTravList = new ArrayList<List<Integer>>();
        
        for(int i=0;i<height;++i){
            levelOrderTravList.add(new ArrayList<Integer>());
        }
        
        populateLevelOrderTrav(root,0,levelOrderTravList);
        
        return levelOrderTravList;
        
        
    }
}